src/doc/book/if-let.md

```diff
@@ -1,7 +1,8 @@
 % if let
 
-`if let` allows you to combine `if` and `let` together to reduce the overhead
-of certain kinds of pattern matches.
+`if let` allows us to match [patterns][patterns] within the condition of an [if][if].
+As a consequence, we reduce the overhead of certain kinds of [pattern][patterns] matches
```

Comment:
After thinking about it more, I'm not a huge fan of the word "consequence" here since it usually describes the results of an action, and there's not really an action here. What about:

> `if let` allows us to reduce...

or

> This feature allows to reduce...

or

> This allows us to reduce...

or something else. What do you think?